File: itlen/app/src/main/java/com/deadely/itl_en/di/component/AppComponent.java
package com.deadely.itl_en.di.component;

import com.deadely.itl_en.di.module.ActivityModule;
import com.deadely.itl_en.di.module.AppModule;
import com.deadely.itl_en.di.module.FragmentModule;

import dagger.Component;

@Component(modules = AppModule.class)
public interface AppComponent {
    ActivityComponent activityComponent(ActivityModule activityModule);

    FragmentComponent fragmentComponent(FragmentModule fragmentModuke);
}

File: itlen/app/src/main/java/com/deadely/itl_en/di/component/AppComponent.kt
package com.deadely.itl_en.di.component

import com.deadely.itl_en.di.module.ActivityModule
import com.deadely.itl_en.di.module.AppModule
import com.deadely.itl_en.di.module.FragmentModule
import dagger.Component

@Component(modules = [AppModule::class])
interface AppComponent {
    fun activityComponent(activityModule: ActivityModule?): ActivityComponent?
    fun fragmentComponent(fragmentModuke: FragmentModule?): FragmentComponent?
}